### DISCUSSION TITLE
Adding bézier curve aided smoothening?

### DISCUSSION THREAD

**Discussion by babaid** [xournalpp/xournalpp#2508] [Ideas]
If this feature would be added, and the pen writing wouldn't be so laggy I would definitively switch to xournalpp forever. Nice work!

**Comment by rolandlo** [comment]
Please provide some specifics about the smoothening algorithm you have in mind in issue #2320. We are currently looking at different algorithms there and try to find the best options.
Pen writing may get less laggy after PR #2339 has been merged.

**Comment by bhennion** [comment]
I think this is not the same issue as #2320, although it is also called smoothing and relates to the way strokes look like. I believe what @babaid is refering to is this kind of things
![2020-12-09-Note-06-44](https://user-images.githubusercontent.com/17818041/101590699-35f0bf80-39eb-11eb-9f5a-21bdf55cd6a9.png)
This kind of angles appear when you don't have enough points (maybe because the input device is to slow or your hand to fast). What could be done (and is done by most softwares I looked at) is to interpolate this stroke using a smooth Bézier curve. With Inkscape, I got
![2020-12-09-Note-06-44-smoothed](https://user-images.githubusercontent.com/17818041/101592242-38a0e400-39ee-11eb-980c-ef7046867765.png)

The reason why I think this enhancement request is different from that of #2320, is that #2320 is about stabilizing the input data to cancel unwanted noise, while this is about what to do with that input data afterwards. I think both are great features that deserve implementation.

I've looked into this Bézier fit a little bit. I'll write up what I found once a feature request has been opened. With the right optimizations, it has other benefits than just the looks of a curve.

**Reply by bhennion** [reply]
> The reason why I think this enhancement request is different from that of #2320, is that #2320 is about stabilizing the input data to cancel unwanted noise, while this is about what to do with that input data afterwards. I think both are great features that deserve implementation.

Based on algorithms like [https://en.wikipedia.org/wiki/Smoothing_spline](https://en.wikipedia.org/wiki/Smoothing_spline), and my work on  #2512, I have changed my mind: Those two issues are more linked than I though. I'll open up a FR to discuss the implications of Bézier fitting.

**Comment by vladQll** [comment]
I would like to know if this functionality was implemented because so far the only way I could correct a little bit was using the arithmetic mean in a value of 15.

**Reply by rolandlo** [reply]
No it's not yet implemented. See  #2729 for work in progress.
